zkvm/src/vm.rs

```diff
@@ -713,44 +712,24 @@ where
     }
 
     fn call(&mut self) -> Result<(), VMError> {
-        // Pop program, blinding factor, contract, and predicate
-        let prog = self.pop_item()?.to_data()?;
-        let blinding = self.pop_item()?.to_data()?;
+        // Pop program, call proof, and contract
+        let program_data = self.pop_item()?.to_data()?;
+        let program = program_data.clone().to_program()?;
```

Comment:
The program is going to be interpreted during the VM execution. We need to preserve the bytecode as-is without parsing it here, as it's going to be parsed again by the VM.

Reply:
The only reason the bytecode is cast to a program is because `prove_taproot` requires that, as documented above. Changes to resolve the above situation should resolve this too.